com.unity.mobile.notifications/Runtime/iOS/Plugins/UnityNotificationManager.m

```diff
@@ -185,7 +191,11 @@ - (void)userNotificationCenter:(UNUserNotificationCenter *)center willPresentNot
     }
 
     if (haveNotificationData)
+    {
+        if (notificationData.userInfo != NULL) // no callback, release user info
```

Comment:
Not sure what `no callback` means, looking through the code, only there is a callback registered, `haveNotificationData` can be YES.

Reply:
userInfo is set to null above in two places after callbacks are invoked. If it's not null at this point, in means there were no callbacks registered and we need to free it manually.

Reply:
But if it's no callbacks, haveNotificationData will be NO, you won't get into this code branch?